### DISCUSSION TITLE
[question] what's the encoding order in `files` or `pieces`? file name by default?

### DISCUSSION THREAD

**Discussion by yanshenxian** [transmission/transmission#2458] [Q&A] [answered]
I try to use `transmission-create` to create a multi-file torrent.
But I don't see any hints about the file order sort with `--help`.
Will it be sorted by file name or ctime by default? Besides, can I specify a sort order?

> System: Ubuntu 16.04.6 LTS
> Cli:  transmission-create 2.84 (14307)

**Comment by ruthtern** [comment, marked as answer]
I just made a torrent with transmission-create. The file order is alphabetical, case-insensitive, numbers sort before letters 
I made the same torrent with mktorrent. The file order is ASCII sort order 
Neither app allows you to specify a different order 
At least one other torrent maker app sorts by file size 
The bittorrent specification does not define a specific order 
The bittorrent specification requires the piece hashes to be created by concatenating the files in the same order as they are in the files list

**Comment by yanshenxian** [comment]
@ruthtern Thanks for your reply~
I have the same files on my pc and server.  I just worried that different encoding order may generate different info_hash...

**Comment by ruthtern** [comment]
The file names are part of the info dictionary, different file name order will create torrent with a different infohash 
I suggest you are overthinking this, inventing a non-existent problem 
Create the torrent once. Use any torrent making app. Send the .torrent file to the other peers
